client/gutenberg/editor/controller.js

```diff
@@ -3,14 +3,28 @@
  * External dependencies
  */
 import React from 'react';
+import { startsWith } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import GutenbergEditor from 'gutenberg/editor/main';
 
+function determinePostType( context ) {
+	if ( startsWith( context.path, '/gutenberg/post' ) ) {
+		return 'post';
+	}
+
+	if ( startsWith( context.path, '/gutenberg/page' ) ) {
+		return 'page';
+	}
+
+	return context.params.type;
```

Comment:
this is a bit opaque here - could we either think about passing `path` and `type` as parameters with docblock comments explaining them or at least add docblock examples, maybe even repeating the route string to make it clear what kind of values to expect for `context.params.type`?